data-model/src/main/java/io/micronaut/data/model/jpa/criteria/impl/query/QueryModelSelectionVisitor.java

```diff
@@ -76,13 +76,13 @@ private QueryModel.Projection getProjection(AggregateExpression<?, ?> aggregateE
         Expression<?> expression = aggregateExpression.getExpression();
         switch (aggregateExpression.getType()) {
             case SUM:
-                return Projections.sum(CriteriaUtils.requireProperty(expression).getProperty().getName());
+                return Projections.sum(CriteriaUtils.requireProperty(expression).getPathAsString());
             case AVG:
```

Comment:
In this case passing `address.street` instead of `street` fixes the issue since property is later being searched in the entity root. If prop is not embedded then there will be no `.` so this should be safe fix.